### ISSUE TITLE
Upgrade from 6.0.17 to 6.0.19 not working

### ISSUE THREAD

**Issue by hos-ftv** [juanluisbaptiste/docker-otrs#68]
Updating from otrs:6.0.17 to otrs:6.0.19 results in `The browser you are using is too old. This software runs with a huge lists of browsers, please upgrade to one of these. Please see the documentation or ask your admin for further information.`

Rolling back to otrs:6.0.17 removes that warning. So I'm not quite sure if it has something to do with OTRS itself or if there are steps missing in this docker-image.

Cleaning caches and sessions via `bin/otrs.Console.pl` or resetting permissions via `bin/otrs.SetPermissions.pl` makes no difference.

So if someone has an idea this would be great.

**Comment by hos-ftv** [comment]
Damn ... seems to be a known bug of otrs: https://bugs.otrs.org/show_bug.cgi?id=14540
So I'll close this issue

**Comment by juanluisbaptiste** [comment]
Yup, found it as soon as I updated the images to 6.0.19, but its already fixed for 6.0.20.

**Comment by hos-ftv** [comment]
This fixed it for me https://bugs.otrs.org/show_bug.cgi?id=14540#c24

But I think you've already seen it ;)

**Comment by hos-ftv** [comment]
A small follow up.
We are still getting the `The browser you are using is too old.` - message after upgrading from 6.0.17 to 6.0.22. This is quite strange.
I already contacted the OTRS - Team but they can't reproduce this behaviour.

Could you please please provide your CommonJS - Settings like you've did [over here](https://bugs.otrs.org/show_bug.cgi?id=14540#c17)? I think that there's something messed up with our installation.

Thanks
Sascha

**Comment by juanluisbaptiste** [comment]
Hi,

Indeed, I had the same issue once upgrading one system from 6.0.19 to 6.0.22, with both the agent and customer interfaces, and I had to apply the same workaround as described on the bug report, using the same CommonJS that was atttached to it to fix it again. For the customer interface the same workaround worked but I had to create another CommonJS settings file.

Try the same file from the bug report and see if it helps, if it doesn't I'll send you mine.

**Comment by hos-ftv** [comment]
Thanks for your answer. I did use the file from the bug-report some weeks ago trying to upgrade to 6.0.20 and 6.0.21. This fixed the issue but bulk-operations stopped working :/

**Comment by juanluisbaptiste** [comment]
Have you tried running scripts/DBUpdate-to-6.pl after the upgrade to see if it fixes the issue ? if the problem really lies here on the container and not on upstream then there must be something that is not being done when doing a minor upgrade. 

Currently on a minor upgrade the only thing done is updating otrs rpm packages to the latest version on the image (doing a rebuild with the latest otrs version) and then pulling the new image and restarting the container, that's how it has worked since I started developing this image while otrs 4.0.x was around.

**Comment by hos-ftv** [comment]
> Have you tried running scripts/DBUpdate-to-6.pl after the upgrade to see if it fixes the issue
Yes. As you can [sea here](https://bugs.otrs.org/show_bug.cgi?id=14691#c4) I ran this script but it resulted in an error/warning. Running this script using 6.0.17 (just using your latest commit and just changing the OTRS version) works as expected.

**Comment by juanluisbaptiste** [comment]
> > Have you tried running scripts/DBUpdate-to-6.pl after the upgrade to see if it fixes the issue
> 
> Yes. As you can [see here](https://bugs.otrs.org/show_bug.cgi?id=14691#c4) I ran this script but it resulted in an error/warning. Running this script using 6.0.17 (just using your latest commit and changing the OTRS version) works as expected.

Sorry, I did not quite understand this. 

You mean that when you run scripts/DBUpdate-to-6.pl on original 6.0.17 image it fails with the error mentioned on the otrs bug report you posted, but if you rebuild a 6.0.17 image using the latest otrs image (currently for 6.0.22) the script runs fine and fixes the issue ?

**Comment by hos-ftv** [comment]
If I use your latest image (or any version from 6.0.19) and run `scripts/DBUpdate-to-6.pl` I get the error mentioned in the  bug-report.
If I fork this project to be up to date with your other changes and just change the OTRS-Version inside the `Dockerfile` to 6.0.17 and build this image on my own, the script works fine.

**Comment by juanluisbaptiste** [comment]
@hos-ftv today I updated the image to OTRS 6.0.23, updated one of my customer installs and did not have this issue. Could yo please test it ?

**Comment by hos-ftv** [comment]
Hi,
sorry for my late reply. I wasn't at the office for a while. Still getting "The browser you are using is to old".
Running `scripts/DBUpdate-to-6.pl` results in:

```
[otrs@93dc1833f736 ~]$ scripts/DBUpdate-to-6.pl

 Migration started ...

 Checking requirements ...

    Requirement check for: Check framework version ...
    Requirement check for: Check required Perl version ...
    Requirement check for: Check required database version ...
    Requirement check for: Check database charset ...
    Requirement check for: Check required Perl modules ...
    Requirement check for: Check if database has been backed up ...

        Did you backup the database? [Y]es/[N]o: Y

    Requirement check for: Upgrade database structure ...
    Requirement check for: Migrating time zone configuration ...
    Requirement check for: Update calendar appointment future tasks ...
    Requirement check for: Migrate GenericAgent jobs configuration ...
    Requirement check for: Migrate TicketAppointment rules configuration ...
    Requirement check for: Create entries in new article table ...
    Requirement check for: Migrate ArticleType in ProcessManagement Data ...
    Requirement check for: Migrate ArticleType in PostMaster filters ...

 Executing tasks ...

    Step 1 of 44: Check framework version ...
    Step 2 of 44: Check required Perl version ...
    Step 3 of 44: Check required database version ...
    Step 4 of 44: Check database charset ...
    Step 5 of 44: Check required Perl modules ...
    Step 6 of 44: Check installed CPAN modules for known vulnerabilities ...
Kernel::System::Console::Command::Dev::Code::CPANAudit could not be loaded: Can't locate Kernel/System/Console/Command/Dev/Code/CPANAudit.pm in @INC (@INC contains: /opt/otrs/Custom /opt/otrs/Kernel/cpan-lib /opt/otrs /usr/local/lib64/perl5 /usr/local/share/perl5 /usr/lib64/perl5/vendor_perl /usr/share/perl5/vendor_perl /usr/lib64/perl5 /usr/share/perl5 .) at /opt/otrs/Kernel/System/ObjectManager.pm line 254.
 at /opt/otrs/scripts/DBUpdateTo6/CPANAuditCheck.pm line 32.
```

**Comment by juanluisbaptiste** [comment]
@hos-ftv now since 6.0.27 the image will detect if the container is running a newer minor version and run `${OTRS_ROOT}/scripts/DBUpdate-to-6.pl --non-interactive`  as instructed by the official documentation. This was done to fix issue #86 , but probably can help here too.

I have not seen this error in a while so I'll close this, reopen if you still have this issue.

**Comment by hos-ftv** [comment]
Hi Juan,

thanks for the advise. As [mentioned previously](https://github.com/juanluisbaptiste/docker-otrs/issues/68#issuecomment-557523458) `${OTRS_ROOT}/scripts/DBUpdate-to-6.pl --non-interactive` throws and Error and stops working. So it does with your current release.

How ever. I dug into it and I found that there was a lot of things mixed up.
At first the `RELEASE` file stated that OTRS 6.0.23 was installed but we were still on 6.0.17.
Than I've checked the missing file `Kernel/System/Console/Command/Dev/Code/CPANAudit.pm` and it is indeed missing from the 6.0.17 release and got introduced some versions later.

So I fixed it by updating OTRS with your latest version 731d979. Then, from within the running container I've downloaded and unzipped OTRS 6.0.27 and copied the whole content (no overwrite) of `Kernel`. After that `${OTRS_ROOT}/scripts/DBUpdate-to-6.pl --non-interactive` worked as it should and I (currently) face no issues.

So I don't know who all this got broken but thanks anyway for your help and effort.

Kind Regards
